window_doc.py

```diff
@@ -0,0 +1,59 @@
+from typing import Union
+
+from talon import Context, Module, actions, ui, app
+
+import os
+
+ctx = Context()
+mod = Module()
+
+@mod.action_class
+class Actions:
+
+    def file_manager_current_path_or_doc():
+        """Returns the current path being viewed/edited, first trying `file_manager_current_path` (in case it's implemented more
+        specifically), then falling back to `window.doc`.
+        """
+
+        path = actions.user.file_manager_current_path()
+        if not path:
+            return ui.active_window().doc.replace("%20", " ")
+
+    def validate_document(doc: str):
```

Comment:
maybe 'represented_file_is_valid'? That's what it's called in code. https://developer.apple.com/documentation/appkit/nswindow/1419066-representedurl

Also need to annotate the return type (here and elsewhere).